### PR TITLE
Fix the PaymentChannel proxy

### LINKED ISSUE #1549: Create "fake" proxy for channel
## Problem Definition
With the old contracts a single `NettingChannel` has an identity on the blockchain and we creates a proxy around this. With the new contracts all channels are managed in the `TokenNetwork`.

To make the refactoring to the new contracts easier we create a fake `Channel` proxy that contains the channel identifier as well as a link to the `TokenNetwork` and provides similar API as the `NettingChannel`.

Related to #1383 

- [x] Create inital Proxy
- [ ] Fix `settlement_period` 
- [ ] Fix `get_all_events` filter
- [ ] Write tests

### PR DESCRIPTION
**Pull request by palango**
- Add `settle_timeout` function back, however this just return the right value as long as the channel is open. See discussion https://github.com/raiden-network/raiden/pull/1551#discussion_r195461907

@hackaugusto To store the correct `settle_timeout` we need to store it ourselves from the `ChannelOpened` event. I can do the changes to the state machine, but that would change the usage of the proxy anyways. 

>> was fixed by adding a filter

Fixes #1549

### CODE REVIEW

File: raiden/tests/integration/contracts/test_payment_channel.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    token_proxy,
+    chain_id,
+    web3,
+):
+    token_network_address = to_canonical_address(token_network_proxy.proxy.contract.address)
+
+    c1_client = JSONRPCClient(
+        '0.0.0.0',
+        blockchain_rpc_ports[0],
+        private_keys[1],
```
Comment (pcppcp) [inline raiden/tests/integration/contracts/test_payment_channel.py]: you'll need
`web3=web`
here

File: raiden/tests/integration/contracts/test_payment_channel.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+from raiden.network.rpc.client import JSONRPCClient
+from raiden.network.proxies import TokenNetwork, PaymentChannel
+from raiden.constants import (
+    NETTINGCHANNEL_SETTLE_TIMEOUT_MIN,
+    EMPTY_HASH,
+)
+
+
+def wait_blocks(web3, blocks):
```
Comment (pcppcp) [inline raiden/tests/integration/contracts/test_payment_channel.py]: this is duplicate of the same function from `test_token_network`, right? Then it should go to i.e. `raiden.tests.utils` to minimize duplicated code.

File: raiden/network/proxies/payment_channel.py [hunk 3]
```diff
@@ -26,11 +29,35 @@ def detail(self) -> Dict:
         """ Returns the channel details. """
         return self.token_network.detail(self.partner_address)
 
-    def settle_block_number(self) -> typing.BlockNumber:
-        """ Returns the settle block of the channel"""
+    def settle_block_number(self) -> int:
+        """ Returns the channels settle block number.
+
+        This is relative while the channel is open and becomes absolute when the channel is closed
```
Comment (pcppcp) [inline raiden/network/proxies/payment_channel.py]: > This is relative while the channel is open and becomes absolute when the channel is closed

this inconsistency can lead to many errors. Does other code depend on the relative number? If absolute block no. can be computed for both open & closed channel, then I'd make the method return it every time.

File: raiden/tests/integration/contracts/test_payment_channel.py (hunk 1 above)
Comment (palango) [reply raiden/tests/integration/contracts/test_payment_channel.py]: Fixed

File: raiden/network/proxies/payment_channel.py (hunk 3 above)
Comment (pcppcp) [reply raiden/network/proxies/payment_channel.py]: you probably can - it's one-liner that can be added later anytime.

**Pull request merged**